specification/configuration.md

```diff
@@ -89,6 +90,52 @@ environment variables:
   If `SPLUNK_CONFIG` is defined then these options are not required but MAY be
   defined.
 
+### Kubernetes
+
+While Kubernetes supports container technology that can be configured using
+environment variables, package management solutions such as Helm charts and
+Operators require YAML-based configuration. As a result, Kubernetes package
+management solutions MUST support the following YAML configuration options:
+
+- `clusterName` ()             : [REQUIRED] Name of the cluster.
+- `cloudProvider` ()           : Where Kubernetes is deployed.
+- `distribution` ()            : Which distribution of Kubernetes is deployed.
+- `environment` ()             : Name of the environment; if not defined then skipped.
+- `agent`                      : Deployed as a DaemonSet.
+  - `enabled` (`true`)         : Whether agent is deployed.
+  - `config` ()                : Updates configuration. Non-list options merged, list options override.
+- `gateway`                    : Deployed as a clustered Service and receives data from agent.
+  - `enabled` (`false`)        : Whether gateway is deployed.
+  - `config` ()                : Updates configuration. Non-list options merged, list options override.
+- `clusterReceiver`            : Deployed as a single replica deployment and collects Kubernetes API cluster and event telemetry.
+  - `enabled` (`true`)         : Whether k8sClusteReceiver is deployed. Ignored if `metricsEnabled` is `false`.
+  - `config` ()                : Updates configuration. Non-list options merged, list options override.
+
+In addition, at least one of the following sections MUST be specified:
+
+- `splunkObservability`
+  - `accessToken` ()           : [REQUIRED] Access token added to exported data.
+  - `realm` ()                 : [REQUIRED] Realm configured for the exporter endpoint.
+  - `logsEnabled` (`false`)    : Whether logs are collected and sent.
+  - `metricsEnabled` (`true`)  : Whether metrics are collected, received, and sent.
+  - `tracesEnabled` (`true`)   : Whether traces are received and sent.
+- `splunkPlatform`
+  - `token` ()                 : [REQUIRED] Token added to exported data.
+  - `endpoint` ()              : [REQUIRED] Where to send exported data.
+  - `logsEnabled` (`true`)     : Whether logs are collected and sent.
+  - `metricsEnabled` (`true`)  : Whether metrics are collected, received, and sent.
+
+Finally, the following Kubernetes secret configuration options MUST be
+supported:
+
+- Splunk Observability
+  - `splunk_observability_access_token`
+- Splunk Platform
+  - `splunk_platform_hec_token`
+  - `splunk_platform_hec_client_cert`
+  - `splunk_platform_hec_client_key`
+  - `splunk_platform_hec_ca_file`
```

Comment:
Should these be interwoven with the above list? They seem subsets of the above configuration options, and potentially duplicates (i.e. `splunk_observability_access_token` looks very similar to `accessToken` in the `splunkObservability` list element)

Also, should they include descriptions and requirement levels?

Reply:
This is specific for k8s secrets. @dmitryax if you specify this do you need to specify say `accessToken` above?

Reply:
> This is specific for k8s secrets. @dmitryax if you specify this do you need to specify say accessToken above?
No, the secrets by default are set from yaml configuration, e.g. `splunkObservability.accessToken` goes to `splunk_observability_access_token`. But, if user wants to specify secrets by themselves, they can set them directly (`splunk_observability_access_token`) -- in that case `splunkObservability.accessToken` can be omitted.

Reply:
> > This is specific for k8s secrets. @dmitryax if you specify this do you need to specify say accessToken above?
> 
> No, the secrets by default are set from yaml configuration, e.g. `splunkObservability.accessToken` goes to `splunk_observability_access_token`. But, if user wants to specify secrets by themselves, they can set them directly (`splunk_observability_access_token`) -- in that case `splunkObservability.accessToken` can be omitted.

Do we need an additional `Kubernetes` section? It seems like this section might be serving two topics? Also, would we not prefer environment variables for native kube configuration of the Splunk observability access token?

Reply:
I updated the previous section to hopefully clarify.